src/Core/Config/Cache/ConfigCacheLoader.php

```diff
@@ -103,47 +103,64 @@ private function loadLegacyConfig($name)
 	{
 		$filePath = $this->baseDir  . DIRECTORY_SEPARATOR . '.' . $name . '.php';
 
+		$config = [];
+
 		if (file_exists($filePath)) {
 			$a = new \stdClass();
 			$a->config = [];
 			include $filePath;
 
+			$htconfigAr = array_keys($a->config);
+
+			// map the legacy configuration structure to the current structure
+			foreach ($htconfigAr as $htconfig) {
+				if (isset($a->config[$htconfig]) && is_array($a->config[$htconfig])) {
```

Comment:
`isset($a->config[$htconfig])` seems to be superfluous as `$htconfig` comes from `array_keys($a->config)` so `$a->config[$htconfig]` will always be set.